src/bootstrap/compile.rs

```diff
@@ -166,7 +161,12 @@ fn copy_self_contained_objects(
     compiler: &Compiler,
     target: Interned<String>,
 ) -> Vec<(PathBuf, DependencyType)> {
-    let libdir = builder.sysroot_libdir(*compiler, target);
+    let libdir_self_contained = if compiler.stage == 0 {
+        builder.sysroot_libdir(*compiler, target).to_path_buf()
+    } else {
+        builder.sysroot_libdir(*compiler, target).join("self_contained")
```

Comment:
```suggestion
        builder.sysroot_libdir(*compiler, target).join("self-contained")
```
Nit: this appears to be more in line with conventions.